### PR TITLE
Databricks operators: added support for pipeline task, jar_params, example DAG for RunNowOperator, and updated docs

### PR DESCRIPTION
**Pull request by robertsaxby**
Added support for pipeline task, corrected documentation urls

### CODE REVIEW

File: airflow/providers/databricks/provider.yaml [hunk 1]
```diff
@@ -22,6 +22,7 @@ description: |
     `Databricks <https://databricks.com/>`__
 
 versions:
+  - 2.0.3
```
Comment (lennartkats-db) [inline airflow/providers/databricks/provider.yaml]: Should this version number follow that from the core Airflow package, i.e. 2.1 or even that of the upcoming 2.2 series? I'm seeing e.g. https://github.com/apache/airflow/pull/17116/files bump another provider to 2.1

**Review by lennartkats-db — APPROVED** [review]
LGTM from Databricks perspective, minor question around version numbers. Also, please note that one test still appears to fail, please review.

File: airflow/providers/databricks/CHANGELOG.rst [hunk 2]
```diff
@@ -19,6 +19,13 @@
 Changelog
 ---------
 
+2.0.3
+.....
+
+Misc
+~~~~
+* ``Updated operator to add support for DLT pipeline, example dags and docs (#19156)``
```
Comment (mik-laj) [inline airflow/providers/databricks/CHANGELOG.rst]: Please remove this change. This file will be updated by release manager.

File: airflow/providers/databricks/provider.yaml [hunk 3]
```diff
@@ -22,6 +22,7 @@ description: |
     `Databricks <https://databricks.com/>`__
 
 versions:
```
Comment (mik-laj) [inline airflow/providers/databricks/provider.yaml]: Please remove this change. This file will be updated by release manager.

File: airflow/providers/databricks/provider.yaml (hunk 1 above)
Comment (mik-laj) [reply airflow/providers/databricks/provider.yaml]: We have separate release cycle for airflow and providers. Version bump will be done by release manager.

File: docs/apache-airflow-providers-databricks/commits.rst [hunk 4]
```diff
@@ -27,6 +27,16 @@ This is detailed commit list of changes for versions provider package: ``databri
 For high-level changelog, see :doc:`package information including changelog <index>`.
 
 
+2.0.3
+.....
+
```
Comment (mik-laj) [inline docs/apache-airflow-providers-databricks/commits.rst]: Please remove this change. This file will be updated by release manager.

File: airflow/providers/databricks/provider.yaml (hunk 1 above)
Comment (robertsaxby) [reply airflow/providers/databricks/provider.yaml]: do you have some details around the provider release cycle and frequency that I could look at? Thank you!

**Comment by alexott** [comment]
`pipeline_task` and `jar_params` where added in #19443.

**Comment by potiuk** [comment]
Needs rebase!

**Pull request closed without merging**